hcmanager/utils_test.go

```diff
@@ -30,43 +30,97 @@ import (
 
 var _ = Describe("HCManager", func() {
 
+	type testCaseFetchBMH struct {
+		namespace     string
+		hosts         []runtime.Object
+		expectedError bool
+	}
+
 	DescribeTable("Test fetch BaremetalHost list",
-		func(namespace string) {
-			hostTest := getHosts()
-			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+		func(tc testCaseFetchBMH) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.hosts...)
 			hcManager := NewHardwareClassificationManager(c, klogr.New())
-			result, _, err := hcManager.FetchBmhHostList(namespace)
-			if err != nil {
+			result, _, err := hcManager.FetchBmhHostList(tc.namespace)
+			if tc.expectedError {
 				Expect(len(result)).To(BeZero())
-			}
-			if len(result) > 0 {
-				Expect(len(result)).Should(Equal(2))
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(result)).NotTo(BeZero())
 			}
 		},
-		Entry("Should fetch BaremetalHosts in ready state and under metal3 namespace", getNamespace()),
-		Entry("Should return error while fetching BaremetalHosts", "sample"),
+		Entry("Should fetch BaremetalHosts in ready state and under metal3 namespace",
+			testCaseFetchBMH{
+				namespace:     getNamespace(),
+				hosts:         getHosts(),
+				expectedError: false},
+		),
+		Entry("Should return error while fetching BaremetalHosts",
+			testCaseFetchBMH{
+				namespace:     "sample",
+				hosts:         getHosts(),
+				expectedError: true},
+		),
 	)
 
+	type testCaseValidation struct {
+		hosts         []runtime.Object
+		hwcProfile    hwcc.HardwareCharacteristics
+		expectedError bool
+	}
+
 	DescribeTable("Test user profiles",
-		func(hwcc hwcc.HardwareCharacteristics) {
-			hostTest := getHosts()
-			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+		func(tc testCaseValidation) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.hosts...)
 			hcManager := NewHardwareClassificationManager(c, klogr.New())
-			err := hcManager.ValidateExtractedHardwareProfile(hwcc)
-			if err != nil {
+			err := hcManager.ValidateExtractedHardwareProfile(tc.hwcProfile)
+			if tc.expectedError {
 				Expect(err).To(HaveOccurred())
 			} else {
-				Expect(err).Should(BeNil())
+				Expect(err).NotTo(HaveOccurred())
 			}
-
 		},
-		Entry("Test Valid Profile", getExtractedHardwareProfile()),
-		Entry("Test Empty profile", getEmptyProfile()),
-		Entry("Test Invalid CPU details", getInvalidCPUProfile()),
-		Entry("Test Invalid DISK details", getInvalidDiskProfile()),
-		Entry("Test Invalid RAM details", getInvalidRAMProfile()),
-		Entry("Test Invalid NIC details", getInvalidNicProfile()),
-		Entry("Test Missing NIC details", getMissingNicDetails()),
+		Entry("Test Valid Profile",
+			testCaseValidation{
+				hosts:         getHosts(),
```

Comment:
if hosts is always the same for all test cases, no need to put it in your struct, it can be just in your function, since it does not change.

Reply:
done. @maelk @digambar15 kindly review.

Reply:
this was not addressed. However, it is a not, so I'm fine to keep it like this if you want it

Reply:
It's already addressed in commit: 0397f0760fd113f5aa217ab4886d36ab4425782e. Kindly have a look.